Lib/traceback.py

```diff
@@ -251,10 +252,12 @@ class FrameSummary:
       mapping the name to the repr() of the variable.
     """
 
-    __slots__ = ('filename', 'lineno', 'name', '_line', 'locals')
+    __slots__ = ('filename', 'lineno', 'end_lineno', 'colno', 'end_colno',
+                 'name', '_line', 'locals')
 
     def __init__(self, filename, lineno, name, *, lookup_line=True,
-            locals=None, line=None):
+            locals=None, line=None,
+            end_lineno=None, colno=None, end_colno=None):
```

Comment:
Let's go consistent with the pre-existing namings (`line`)
```suggestion
            end_line=None, offset=None, end_offset=None):
```

Reply:
Oh, `line` here is the actual string for the line. There is a separate `lineno` attribute that is the line number.